web/src/test/java/com/epam/rft/atsy/web/controllers/NewApplicationControllerTest.java

```diff
@@ -1,152 +1,235 @@
+
+    verify(applicationsService).saveApplication(applicationDTOCaptor.capture(), stateHistoryDTOCaptor.capture());
+    then(applicationsService).should()
+        .saveApplication(applicationDTOCaptor.getValue(), stateHistoryDTOCaptor.getValue());
+    then(this.candidateService).should().getCandidate(1L);
+    then(this.positionService).should().getPositionDtoById(1L);
+    then(this.channelService).should().getChannelDtoById(1L);
+
+    assertThat(stateHistoryDTOCaptor.getValue(), equalTo(stateHistoryDTO));
+    assertApplicationDtoWhenSaveOrUpdateIsSuccess(applicationDTOCaptor);
+  }
+
+  @Test
+  public void saveOrUpdateShouldBeUnsuccessfulWhenCandidateIdIsNotANumber() throws Exception {
+
+    this.mockMvc.perform(post(REQUEST_URL_POST)
+        .param("candidateId", TEXT)
+        .param("position.id", STRING_VALUE_ONE).param("position.name", POSITION_NAME_DEVELOPER)
+        .param("channel.id", STRING_VALUE_ONE).param("channel.name", CHANNEL_NAME_FACEBOOK)
+        .param("description", DESCRIPTION))
+        .andExpect(status().is3xxRedirection())
+        .andExpect(redirectedUrl(REDIRECT_URL_FOR_NON_EXISTING_CANDIDATE));
+
+    verifyZeroInteractions(applicationsService);
+    then(this.candidateService).should().getCandidate(null);
+    then(this.positionService).should().getPositionDtoById(1L);
+    then(this.channelService).should().getChannelDtoById(1L);
+  }
+
+  @Test
+  public void saveOrUpdateShouldBeUnsuccessfulWhenCandidateIdNotExists() throws Exception {
+    given(this.messageKeyResolver.resolveMessageOrDefault(COMMON_INVALID_INPUT_MESSAGE_KEY)).willReturn(COMMON_INVALID_INPUT_MESSAGE);
+    given(this.messageKeyResolver.resolveMessageOrDefault(CANDIDATE_NOT_FOUND_ERROR_MESSAGE_KEY)).willReturn(CANDIDATE_NOT_FOUND_ERROR_MESSAGE);
+
+    this.mockMvc.perform(post(REQUEST_URL_POST)
+        .param("candidateId", STRING_VALUE_TWO)
+        .param("position.id", STRING_VALUE_ONE).param("position.name", POSITION_NAME_DEVELOPER)
+        .param("channel.id", STRING_VALUE_ONE).param("channel.name", CHANNEL_NAME_FACEBOOK)
+        .param("description", DESCRIPTION))
+        .andExpect(status().is3xxRedirection())
+        .andExpect(flash().attribute("candidateIdErrorMessage", CANDIDATE_NOT_FOUND_ERROR_MESSAGE))
+        .andExpect(redirectedUrl(REDIRECT_URL_FOR_CANDIDATE_WITH_WRONG_ID));
+
+    verifyZeroInteractions(applicationsService);
+    then(this.candidateService).should().getCandidate(2L);
+    then(this.positionService).should().getPositionDtoById(1L);
+    then(this.channelService).should().getChannelDtoById(1L);
+  }
+
+  @Test
+  public void saveOrUpdateShouldBeUnsuccessfulWhenPositionIdIsNotANumber() throws Exception {
+    given(this.messageKeyResolver.resolveMessageOrDefault(COMMON_INVALID_INPUT_MESSAGE_KEY)).willReturn(COMMON_INVALID_INPUT_MESSAGE);
+    given(this.messageKeyResolver.resolveMessageOrDefault(POSITION_NOT_FOUND_ERROR_MESSAGE_KEY)).willReturn(POSITION_NOT_FOUND_ERROR_MESSAGE);
+

```

Comment:
There is no `given` initialization for the `candidateService`, `positionService` and the `channelService`.